crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -1512,9 +1517,8 @@ pub struct EntityEntryCommands<'a, T> {
 
 impl<'a, T: Component> EntityEntryCommands<'a, T> {
     /// Modify the component `T` if it exists, using the the function `modify`.
-    pub fn and_modify(mut self, modify: impl FnOnce(Mut<T>) + Send + Sync + 'static) -> Self {
-        self.entity_commands = self
-            .entity_commands
+    pub fn and_modify(&mut self, modify: impl FnOnce(Mut<T>) + Send + Sync + 'static) -> &mut Self {
```

Comment:
I think this makes sense, but I want to highlight that the changes to `EntityEntryCommands` aren't exactly a revert.  This type didn't exist at the time of the original change, and it *could* make sense for it to work differently from `EntityCommands`.

Reply:
That is true, I should have thought to mention this. Sorry about that